cirq-google/cirq_google/devices/grid_device.py

```diff
@@ -619,25 +633,46 @@ def validate_operation(self, operation: cirq.Operation) -> None:
         Raises:
             ValueError: The operation isn't valid for this device.
         """
+        return self._validate_operations(iter([operation]))
+
+    def _validate_operations(self, operations: Iterator[cirq.Operation]) -> None:
+        """Raises an exception if any of the operations are not valid.
+
+        An operation is valid if
+            * The operation is in the device gateset.
+            * The operation targets a valid qubit
+            * The operation targets a valid qubit pair, if it is a two-qubit operation.
+
+        Args:
+            operation: The operation to validate.
 
-        if operation not in self._metadata.gateset:
-            raise ValueError(f'Operation {operation} contains a gate which is not supported.')
-
-        for q in operation.qubits:
-            if isinstance(q, ops.Coupler):
-                if any(qc not in self._metadata.qubit_set for qc in q.qubits):
-                    raise ValueError(f'Qubits on coupler not on device: {q.qubits}.')
-                if frozenset(q.qubits) not in self._metadata.qubit_pairs:
-                    raise ValueError(f'Coupler pair is not valid on device: {q.qubits}.')
-            elif q not in self._metadata.qubit_set:
-                raise ValueError(f'Qubit not on device: {q!r}.')
-
-        if (
-            len(operation.qubits) == 2
-            and not any(operation in gf for gf in _VARIADIC_GATE_FAMILIES)
-            and frozenset(operation.qubits) not in self._metadata.qubit_pairs
-        ):
-            raise ValueError(f'Qubit pair is not valid on device: {operation.qubits!r}.')
+        Raises:
+            ValueError: The operation isn't valid for this device.
+        """
+        gateset = self._metadata.gateset
+        qubit_pairs = self._metadata.qubit_pairs
+        qubits = self._metadata.qubit_set
+        for operation in operations:
+            if operation not in gateset:
+                raise ValueError(f'Operation {operation} contains a gate which is not supported.')
+
+            for q in operation.qubits:
+                if isinstance(q, ops.Coupler):
+                    if any(qc not in qubits for qc in q.qubits):
+                        raise ValueError(f'Qubits on coupler not on device: {q.qubits}.')
+                    if frozenset(q.qubits) not in qubit_pairs:
+                        raise ValueError(f'Coupler pair is not valid on device: {q.qubits}.')
+                elif q not in qubits:
+                    raise ValueError(f'Qubit not on device: {q!r}.')
+
+            if (
+                len(operation.qubits) == 2
+                and not any(
+                    isinstance(operation.gate, gate_type) for gate_type in _VARIADIC_GATE_TYPES
+                )
```

Comment:
```suggestion
                and not isinstance(operation.gate, _VARIADIC_GATE_TYPES)
```

after changing `_VARIADIC_GATE_TYPES` to a tuple.